src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobMetadataStore.cs

```diff
@@ -130,12 +129,14 @@ public Task<DicomDataset> GetInstanceMetadataAsync(VersionedInstanceIdentifier v
 
         return ExecuteAsync(async t =>
         {
-            await using (Stream stream = _recyclableMemoryStreamManager.GetStream(GetInstanceMetadataStreamTagName))
-            {
-                await blobClient.DownloadToAsync(stream, cancellationToken);
+            BlobDownloadResult result = await blobClient.DownloadContentAsync(t);
 
+            // DICOM metadata file was saved using UTF-8 encoding with BOM, When using JsonSerializer, by design UTF-8 BOM should be ignored
+            // Else, the following exception will occur "System.Text.Json.JsonReaderException: ''0xEF' is an invalid start of a value. LineNumber: 0 | BytePositionInLine: 0." at the line Reader.Read.
+            // So reading the content into stream which will remove the UTF-8 BOM. https://github.com/dotnet/runtime/issues/29838
+            using (Stream stream = _recyclableMemoryStreamManager.GetStream(result.Content))
+            {
                 stream.Seek(0, SeekOrigin.Begin);
-
                 return await JsonSerializer.DeserializeAsync<DicomDataset>(stream, _jsonSerializerOptions, t);
```

Comment:
Shouldn't the `Stream`-based APIs automatically handle the BOM? The linked issue in dotnet/runtime seems to indicate this too

Reply:
I just loaded a UTF8-BOM file locally, and can see the `JsonSerializer` method parses just fine:
```csharp
using Stream file = File.OpenRead(@"...");
JsonElement element = JsonSerializer.Deserialize<JsonElement>(file);
```

Reply:
Oh, I see! You are changing it back. The reason it didn't work is indeed because of the BOM, but it is because of the API we used: [`ToObjectFromJson`](https://github.com/Azure/azure-sdk-for-net/blob/d46247724a5467a71269f2bbb80811f7c1838178/sdk/core/System.Memory.Data/src/BinaryData.cs#L257). Looking at the implementation, it attempts to deserialize the bytes directly which does not process the BOM unlike the `Stream` overload.

Reply:
Yeah, the deserialization was the issue. Passing stream directly